### PR TITLE
 spatial/r3: add (Vec).Cross

### PR DESCRIPTION
**Pull request by odsod**
https://github.com/gonum/gonum/issues/1049

Submitting this change as an example of operations to add. Happy to provide more common operations in subsequent PRs if this one gets accepted.

### CODE REVIEW

File: spatial/r3/vector_test.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package r3
+
+import "testing"
+
+func TestVector_Cross(t *testing.T) {
+	a := Vec{X: 2, Y: 3, Z: 4}
+	b := Vec{X: 5, Y: 6, Z: 7}
+	expected := Vec{X: -3, Y: 6, Z: -3}
+	actual := a.Cross(b)
```
Comment (sbinet) [inline spatial/r3/vector_test.go]: and that `got`:
```suggestion
	got := a.Cross(b)
```

File: spatial/r3/vector_test.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package r3
```
Comment (sbinet) [inline spatial/r3/vector_test.go]: please add the missing copyright header.

File: spatial/r3/vector.go [hunk 3]
```diff
@@ -17,6 +17,15 @@ func (p Vec) Add(q Vec) Vec {
 	return p
 }
 
+// Cross returns the vector cross product of p and q.
```
Comment (sbinet) [inline spatial/r3/vector.go]: we could perhaps replace `and` with the cross symbol (a simple `x` would do, but we could also reach for UTF8's `×`)

**Comment by odsod** [comment]
@sbinet Thanks for the feedback, fixes coming shortly!

File: spatial/r3/vector.go (hunk 3 above)
Comment (kortschak) [reply spatial/r3/vector.go]: The UTF-8 symbol is what we usually use where we use a symbol, though we often don't use a symbol.

**Comment by odsod** [comment]
Update - we moved in a different direction and are not needing this functionality - dropping this PR. Cheers

**Pull request closed without merging**